### PR TITLE
better order of columns in fields tab (vector layer properties)

### PR DESCRIPTION
**Pull request by 3nids**
This moves to the left edit type and alias columns in the fields tab.
They used to be placed further (after comment) and one has to played with scroll bars very often.
These are the most used columns (at least for me), so I think they should  be moved to the left.

Can this be considered as a bugfix???

image

### CODE REVIEW

**Comment by m-kuhn** [comment]
YES!

File: src/app/qgsfieldsproperties.cpp [hunk 1]
```diff
@@ -305,7 +305,14 @@ void QgsFieldsProperties::setRow( int row, int idx, const QgsField& field )
...
 
-  for ( int i = 0; i < attrCommentCol; i++ )
+  QList<int> notEditableCols = QList<int>()
+                               << attrIdCol
+                               << attrNameCol
+                               << attrTypeCol
+                               << attrTypeNameCol
+                               << attrLengthCol
+                               << attrPrecCol;
+  Q_FOREACH ( int i, notEditableCols )
```
Comment (m-kuhn) [inline src/app/qgsfieldsproperties.cpp]: What's the reason for this change? Doesn't this just duplicate the order information which is already encoded in the enum (and therefore increase the risk that somebody forgets to change one of the two)?

Comment (3nids) [inline src/app/qgsfieldsproperties.cpp]: Actually it's other way around. 
The old code was considering the order of columns fixed. The new one considers individual columns which should be set as read only.

Comment (m-kuhn) [inline src/app/qgsfieldsproperties.cpp]: Ah, I get it, sorry

**Comment by 3nids** [comment]
Thanks

**Comment by m-kuhn** [comment]
As a friendly reminder ;)

https://github.com/qgis/QGIS-Enhancement-Proposals/issues/37
